ipa-core/src/protocol/ipa_prf/prf_sharding/mod.rs

```diff
@@ -43,6 +51,40 @@ pub struct PrfShardedIpaInputRow<BK: SharedValue, TV: SharedValue, TS: SharedVal
     pub breakdown_key: Replicated<BK>,
     pub trigger_value: Replicated<TV>,
     pub timestamp: Replicated<TS>,
+    pub sort_key: Replicated<BA32>,
+}
+
+impl<BK: SharedValue, TS, TV: SharedValue> PrfShardedIpaInputRow<BK, TV, TS>
+where
+    TS: SharedValue + ArrayAccess<Output = Boolean> + Expand<Input = Boolean>,
+{
+    /// This function defines the sort key.
+    /// The order of sorting is `timestamp`, `is_trigger_bit`, `counter`.
+    /// We sort by `is_trigger_bit` to ensure source events come before trigger in case there
+    /// is a tie in timestamp
+    /// Counter is added to ensure each sorting key is unique to avoid privacy leakage
+    /// NOTE: the order of members to be sorted by
+    /// should be opposite to what you want as the bits are stored in Little Endian format.
+    /// We still need to add epoch which will be added later
+    pub fn compute_sort_key(&mut self, counter: u64) {
+        let mut y: Replicated<BA32> = Replicated::ZERO;
+
+        expand_shared_array_in_place(
+            &mut y,
+            &Replicated::new(BA7::truncate_from(counter), BA7::truncate_from(counter)),
+            0,
+        );
+        let mut offset = BA7::BITS as usize;
+
+        y.0.set(offset, self.is_trigger_bit.left());
+        y.1.set(offset, self.is_trigger_bit.right());
+
+        offset += 1;
+        expand_shared_array_in_place(&mut y, &self.timestamp, offset);
+        // TODO(richaj): add epoch to sort key computation
+
+        self.sort_key = y;
```

Comment:
OK, this is pretty cool. Only downside is that we are now storing the information about the timestamp, epoch, and is_trigger_bit in two places on this struct. In a future PR we can probably consolidate this and only store it once.